pkg/cli/helm/cluster.go

```diff
@@ -298,16 +298,38 @@ func (i *Impl) CheckRadiusInstall(kubeContext string) (InstallState, error) {
 	// Check if Radius is installed
 	radiusInstalled, radiusVersion, err := helmAction.QueryRelease(kubeContext, clusterOptions.Radius.ReleaseName, clusterOptions.Radius.Namespace)
 	if err != nil {
-		return InstallState{}, err
+		return InstallState{}, fmt.Errorf("failed to check Radius installation (release: %s, namespace: %s, context: %s): %w",
+			clusterOptions.Radius.ReleaseName, clusterOptions.Radius.Namespace, kubeContext, err)
+	}
+
+	// If Radius is installed, check if we're in an upgrade scenario
+	// During a Helm upgrade, the release is in a pending state with the new version already set
+	// We need to get the previous deployed version instead
+	if radiusInstalled && radiusVersion != "" {
```

Comment:
when would radius be installed but no specific version detected? wouldn't radiusVersion always have a value?

Reply:
@sk593 You're right that normally radiusVersion should always have a value if Radius is installed. The check for `radiusVersion != ""` is defensive programming to handle edge cases like corrupted Helm releases or incomplete metadata. I've added a comment to clarify this in the code.